packages/react-bootstrap-table2-paginator/src/wrapper.js

```diff
@@ -0,0 +1,99 @@
+/* eslint react/prop-types: 0 */
+/* eslint arrow-body-style: 0 */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+const wrapperFactory = (baseElement, Const) =>
+  class PaginationWrapper extends Component {
+    static propTypes = {
+      store: PropTypes.object.isRequired,
+      options: PropTypes.object
+    }
+
+    static defaultProps = {
+      options: {
+        paginationSize: Const.PAGINATION_SIZE,
+        pageStartIndex: Const.PAGE_START_INDEX,
+        withFirstAndLast: Const.With_FIRST_AND_LAST,
+        alwaysShowAllBtns: Const.SHOW_ALL_PAGE_BTNS,
+        firstPageText: Const.FIRST_PAGE_TEXT,
+        prePageText: Const.PRE_PAGE_TEXT,
+        nextPageText: Const.NEXT_PAGE_TEXT,
+        lastPageText: Const.LAST_PAGE_TEXT,
+        sizePerPageList: Const.SIZE_PER_PAGE_LIST,
+        onSizePerPageChange: undefined,
+        onPageChange: undefined
+      }
+    }
+
+    constructor(props) {
+      super(props);
+      const options = props.options || {};
+      const currPage = options.pageStartIndex || Const.PAGE_START_INDEX;
+      this.handleChangePage = this.handleChangePage.bind(this);
+      this.handleChangeSizePerPage = this.handleChangeSizePerPage.bind(this);
+      this.state = {
+        currPage,
+        currSizePerPage: Const.SIZE_PER_PAGE_LIST[0]
+      };
+    }
+
+    handleChangePage(currPage) {
+      const { options } = this.props;
+      if (options.onPageChange) {
+        options.onPageChange(currPage, this.state.currSizePerPage);
+      }
+      this.setState(() => {
+        return {
+          currPage
+        };
+      });
+    }
+
+    handleChangeSizePerPage(currSizePerPage, currPage) {
+      const { options } = this.props;
+      if (options.onSizePerPageChange) {
+        options.onSizePerPageChange(currSizePerPage, currPage);
+      }
+      this.setState(() => {
+        return {
+          currPage,
+          currSizePerPage
+        };
+      });
+    }
+
+    render() {
+      const { pagination: { Pagination }, options, store } = this.props;
+      const { currPage, currSizePerPage } = this.state;
+      const base = baseElement({
+        ...this.props,
+        key: 'table',
+        data: store.getByCurrPage(currPage, currSizePerPage)
+      });
+
+      return [
```

Comment:
there's a kind of different than previous wrapper, 
1. In this class, I export a factory function for creating pagination wrapper
2. This factory function will have two args: `baseElement` and `Const` which are passed from `react-bootstrap-table2`(`table-factory`)

3. This HOC component is not render next element, instead, I render next element and Pagination component together